map/mwm_url.cpp

```diff
@@ -93,38 +90,40 @@ bool ParseLatLon(std::string const & key, std::string const & value,
 
 // A helper for SetUrlAndParse() below.
 // kGe0Prefixes supports API and ge0 links, kLegacyMwmPrefixes - only API.
-inline tuple<size_t, bool> findUrlPrefix(string const & url)
+std::tuple<size_t, bool> FindUrlPrefix(std::string const & url)
 {
-  for (auto prefix : ge0::kGe0Prefixes)
+  for (auto const prefix : ge0::Ge0Parser::kGe0Prefixes)
   {
     if (strings::StartsWith(url, prefix))
       return {prefix.size(), true};
   }
-  for (auto prefix : kLegacyMwmPrefixes)
+  for (auto const prefix : kLegacyMwmPrefixes)
   {
     if (strings::StartsWith(url, prefix))
       return {prefix.size(), false};
   }
-  return {string::npos, false};
+  return {std::string::npos, false};
 }
 }  // namespace
 
-ParsedMapApi::UrlType ParsedMapApi::SetUrlAndParse(string const & raw)
+ParsedMapApi::UrlType ParsedMapApi::SetUrlAndParse(std::string const & raw)
 {
   Reset();
-  SCOPE_GUARD(guard, [this]() { Reset(); });
+  SCOPE_GUARD(guard, [this]
+  {
+    if (m_requestType == UrlType::Incorrect)
+      Reset();
+  });
 
-  auto [prefix, checkForGe0Link] = findUrlPrefix(raw);
-  if (prefix != string::npos)
+  if (auto const [prefix, checkForGe0Link] = FindUrlPrefix(raw); prefix != std::string::npos)
   {
-    url::Url const url = url::Url("om://" + raw.substr(prefix));
+    url::Url const url {"om://" + raw.substr(prefix)};
```

Comment:
Because {} constructors explicitly check the type passed in and fail during the compilation if the type is implicitly converted.

In simple words: https://www.modernescpp.com/index.php/initialization/

Some pros are listed [here](https://stackoverflow.com/a/18222927)

Details [here](https://en.cppreference.com/w/cpp/language/aggregate_initialization).

There are many other reasons, with only a few exceptions when ` = ` should be used.